graphics/apitrace/Portfile

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem             1.0
+
+PortGroup               cmake 1.1
+PortGroup               github 1.0
+
+categories              graphics devel
+license                 MIT
+maintainers             nomantainer openmaintainer
+
+description             OpenGL API inspector
+long_description        apitrace is a free program with tools to inspect and trace OpenGL api calls.
+
+github.setup            apitrace apitrace 13.0
+github.tarball_from     tarball
+fetch.type              git
+
+supported_archs         x86_64 i386
+
+configure.args-append   -DENABLE_X11=OFF \
+                        -DBUILD_TESTING=OFF \
+                        -DENABLE_GUI=OFF
+
+checksums               rmd160  81cf9940f583cfcb07e98c55ca3a3e841045feab \
+                        sha256  f885916d94d896cd67795e5e5c871c3b3d04b5fbfae2ff8450daac6db2f5aa7f \
+                        size    2274896
+
+depends_lib-append      port:brotli \
+                        port:libpng \
+                        port:python312 \
+                        port:py312-Pillow \
+                        port:zlib
+
+patchfiles-append       do-not-fail-submodules.diff \
+                        patch-CMakeLists.diff
+
+variant qtgui description {Build qt5 GUI} {
+    PortGroup               qt5 1.0
+
+    configure.args-replace -DENABLE_GUI=OFF -DENABLE_GUI=ON
+}
+
+variant tests description {Build tests} {
+    test.run                    yes
+    depends_build-append        port:gtest
+    configure.args-replace      -DBUILD_TESTING=OFF -DBUILD_TESTING=ON
+}
+
+# Maybe some other day I'll get it working. Honestly, I doubt many people are gonna use this port, let alone the extended test suite.
+
+#subport apitrace-tests {
+#   description                 Build and run the apitrace extra testsuite
+#   patchfiles                  patch-tests-CMakeLists.diff
+#   github.setup                apitrace apitrace-tests b0a8664
+#   test.run
+#   depends_lib-append          port:glfw \
+#                               port:gtest
+#   checksums           rmd160  7790eb5789cf1283baeec3e5619d894e43d6160b \
+#                       sha256  b23937cfd7910da1a470b2b7ba601daf9f7410618514baa77fca202a22d5dac5 \
+#                       size    708985
+#}
+
+if { ${name} eq ${subport} } {
+    post-fetch {
+        system -W ${worksrcpath} "git submodule update --init --remote thirdparty/snappy"
+    }
```

Comment:
can "snappy" installed by MacPorts used instead of this?

Reply:
Possibly it could be done by patching the portfile. It's the only of the bundled third-party things I left, and I did it because of this:

https://github.com/apitrace/apitrace/blob/45a005875d348b055d5b88dca285c109dee90457/docs/INSTALL.markdown?plain=1#L34-L36